### PR TITLE
Fix REST API

### PR DESCRIPTION
**Pull request by Selameab**
The TruthValue constructor accepts (strength, confidence) not (strength, count). Count has been converted to confidence whenever necessary.

self.atomspace = opencog.cogserver.get_server_atomspace() has been commented because it was causing this error: "terminate called after throwing an instance of 'opencog::AssertionException' what():  Accidentally called the base class!"

### CODE REVIEW

**Comment by AmeBel** [comment]
Hi, the RestApi Test are still failing

**Comment by Selameab** [comment]
It's passing on my container. Does this use the latest atomspace?

File: opencog/python/web/api/apiatomcollection.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __init__(self):
         self.reqparse.add_argument('limit', type=int, location='args')
 
         super(AtomCollectionAPI, self).__init__()
-        self.atomspace = opencog.cogserver.get_server_atomspace()
+        # self.atomspace = opencog.cogserver.get_server_atomspace()
```
Comment (AmeBel) [inline opencog/python/web/api/apiatomcollection.py]: Why did you comment out this line? If it doesn't use the default cogserver atomspace then how will one be able to access the content in the default cogserver-atomspace?

**Comment by AmeBel** [comment]
@Selameab the tests pass. Have you been able to run it from the opencog shell? I get
```
opencog> restapi.Start
command "restapi.Start" not found
Available commands:
....
opencog> loadpy restapi
Error: The opencog.agent_finder module is not loaded; Cannot load other python modules as a result.opencog> 
```
I was thinking of testing it with https://github.com/opencog/external-tools/tree/master/ocViewer to confirm the default cogserver's atomspace content are being accessed, as the their are no tests that run against the cogserver.

It would be great if you could add a test against cogserver, so as to avoid manual testing.

**Comment by Selameab** [comment]
You have to follow these instructions to get it to work...
https://github.com/opencog/opencog/blob/master/opencog/python/README.md

Changes made inside the scheme shell are reflected in the ocViewer. So, yes the restapi is using the cogserver's atomspace...

**Comment by AmeBel** [comment]
I have followed that step

**Comment by Selameab** [comment]
Is restapi.Start available in the OpenCog shell?
Are you using the default cogserver configurations?

**Comment by AmeBel** [comment]
no restapi.Start isn't available
yes the default configuration

If I am not mistaken you are running it inside a docker container right?

File: opencog/python/web/api/apiatomcollection.py (hunk 1 above)
Comment (Selameab) [inline opencog/python/web/api/apiatomcollection.py]: That won't be necessary. Passing an atomspace to the constructor seems like the optimal way to do it but AtomCollectionAPI receives the default AtomSpace through it's new class method. Cosmo has explained why he chose this approach: https://github.com/twilio/flask-restful/issues/134

**Comment by Selameab** [comment]
Yes, I'm running it inside the docker container with port 5000 exposed to the host.

Is the environment variable $PYTHONPATH equal to "/usr/local/share/opencog/python:/opencog/opencog/python:/opencog/build/opencog/cython?"

**Comment by AmeBel** [comment]
yap

**Comment by Selameab** [comment]
As a last resort, try restarting your container (not just the cogserver). The environment vars might have not propagated...

**Comment by AmeBel** [comment]
did that, also i built after removing all the content in my build directory.

**Comment by AmeBel** [comment]
@pennachin I am having this https://github.com/opencog/opencog/pull/1770#issuecomment-140625465 issue, tests pass. If you can would you please test the pr, and check if you could start the restapi. 

Thanks

**Comment by pennachin** [comment]
Sure, I'll take a look, but it will be couple of days. I'm in the middle of
a long travel leg...

On Wednesday, September 16, 2015, Amen Belayneh <notifications@github.com>
wrote:

> @pennachin <https://github.com/pennachin> I am having this #1770 (comment)
> <https://github.com/opencog/opencog/pull/1770#issuecomment-140625465>
> issue, tests pass. If you can would you please test the pr, and check if
> you could start the restapi.
>
> Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencog/opencog/pull/1770#issuecomment-140701333>.
>


-- 
Cassio Pennachin

**Comment by AmeBel** [comment]
@Selameab Why did you disable the AnaphoraTest

**Comment by Selameab** [comment]
Cassio mentioned it in an email about a month ago...

**Comment by AmeBel** [comment]
I don't know in what context Cassio suggested disabling it, but we don't
want to disable tests. Fixing them is prefered.

**Comment by pennachin** [comment]
I did mention that privately in late August, but it was before changes made by @linas in September. Linas says he can run those tests by hand (although I have failed to do so, as have others), so I think we should try one more time to debug that test instead of disabling it.

@AmeBel I'm sorry I haven't had time to investigate this issue, but I'll be at PolyU all afternoon today and we can take a look together if you have some time.

**Comment by AmeBel** [comment]
@Selameab pls merge recent changes from master for merging.

**Comment by pennachin** [comment]
I've tested this and everything works for me. However we should NOT merge the commit that disables the AnaphoraUTest.

**Comment by Selameab** [comment]
@pennachin I've removed the commit that disables the AnaphoraUTest.

**Comment by pennachin** [comment]
Thanks!

**Pull request merged**